src/commcare_cloud/commands/ansible/service.py

```diff
@@ -1,328 +1,443 @@
-                "For Celery, this can be Celery worker names as appearing in app-processes.yml. "
-                "This can also be a comma-separated list of workers"
-                "For ex: Run 'commcare-cloud staging celery status --only=sms_queue,email_queue'"
+
+
+class SubServicesMixin(six.with_metaclass(ABCMeta)):
+    @abstractproperty
+    def managed_services(self):
+        raise NotImplementedError
+
+    def get_options(self):
+        options = super(SubServicesMixin, self).get_options()
+        managed_services = self.managed_services
+        if managed_services:
+            options["Sub-services (use with --only)"] = [ServiceOption(service) for service in managed_services]
+        return options
+
+
+class SupervisorService(SubServicesMixin, ServiceBase):
+    def execute_action(self, action, host_pattern=None, process_pattern=None):
+        if host_pattern:
+            self.environment.inventory_manager.subset(host_pattern)
+
+        process_host_mapping = self._get_processes_by_host(process_pattern)
+
+        if not process_host_mapping:
+            raise NoHostsMatch
+
+        non_zero_exits = []
+        for hosts, processes in process_host_mapping.items():
+            command = 'supervisorctl {} {}'.format(
+                action,
+                ' '.join(processes)
             )
-        )
+            exit_code = self._run_ansible_module(
+                ','.join(hosts),
+                'shell',
+                command
+            )
+            if exit_code != 0:
+                non_zero_exits.append(exit_code)
+        return non_zero_exits[0] if non_zero_exits else 0
+
+    @abstractmethod
+    def _get_processes_by_host(self, process_pattern=None):
+        """
+        :param process_pattern: process pattern from the args or None
+        :return: dict mapping tuple(hostname1,hostname2,...) -> [process name list]
+        """
+        raise NotImplemented
+
+
+class AnsibleService(ServiceBase):
+    """Service that is controlled via the ansible 'service' module"""
+
+    @property
+    def service_name(self):
+        return self.name
+
+    def execute_action(self, action, host_pattern=None, process_pattern=None):
+        if action == 'status':
+            host_pattern = host_pattern or ','.join(self.inventory_groups)
+            command = 'service {} status'.format(self.service_name)
+            return self._run_ansible_module(host_pattern, 'shell', command)
+
+        host_pattern = host_pattern or ','.join(self.inventory_groups)
+        service_args = 'name={} state={}'.format(self.service_name, STATES[action])
+        return self._run_ansible_module(host_pattern, 'service', service_args)
 
-    def get_inventory_group_for_service(self, service, service_group):
-        return self.INVENTORY_GROUP_FOR_SERVICE.get(service, service_group)
 
-    @staticmethod
-    def get_celery_workers_to_work_on(args):
+class MultiAnsibleService(SubServicesMixin, AnsibleService):
```

Comment:
@dannyroberts @mkangia I was thinking about removing this class. It's only used RiakCS and Kafka but the same functionality can be accomplished by passing multiple services to the command line and it adds quite a lot of code:

I also don't think it's that common that you want to manage these services as a single unit. 

```
# current
cchq <env> service riakcs restart
cchq <env> service riakcs restart --only riakcs

# If we split them up
cchq <env> service riakcs riak stanchion restart
cchq <env> service riakcs restart
```